src/style.js

```diff
@@ -356,11 +350,17 @@ const getFrameClip = memoizeClip((clipPath, context, dimensions) => {
     .attr("height", height - marginTop - marginBottom);
 });
 
-const getProjectionClip = memoizeClip((clipPath, context) => {
-  const {projection} = context;
-  if (!projection) throw new Error(`the "sphere" clip option requires a projection`);
-  clipPath.append("path").attr("d", geoPath(projection)({type: "Sphere"}));
-});
+function memoizeGeo(clip) {
+  const geoClips = new WeakMap();
+  return (geo, scales) => {
+    if (!geoClips.has(geo)) geoClips.set(geo, clip(geo, scales));
+    return geoClips.get(geo);
+  };
+}
+
+const getGeoClip = memoizeGeo((geo) =>
+  memoizeClip((clipPath, context) => clipPath.append("path").attr("d", context.path()(geo)))
+);
```

Comment:
I feel like there’s a way to simplify this memoization logic… but I’m just polishing. I’ll look at it if I’m inspired.

Reply:
I agree that it feels convoluted, like a factory of factory… I didn't find a simpler way yet, but I'm sure there must be one.

Reply:
done!